src/p2p/net_node.inl

```diff
@@ -1095,19 +1161,22 @@ namespace nodetool
     {
       if(conn_count < expected_white_connections)
       {
-        //start from white list
-        if(!make_expected_connections_count(true, expected_white_connections))
+        //start from anchor list
+        if(!make_expected_connections_count(anchor, P2P_DEFAULT_ANCHOR_CONNECTIONS_COUNT))
           return false;
-        //and then do grey list
-        if(!make_expected_connections_count(false, m_config.m_net_config.connections_count))
+        //then do white list
+        if(!make_expected_connections_count(white, expected_white_connections))
```

Comment:
I think this should be expected_white_connections + P2P_DEFAULT_ANCHOR_CONNECTIONS_COUNT, unless you're intending to have less white peers with this patch. 263.pdf says "two extra outgoing connections".